client/verta/verta/deployment/_deployedmodel.py

```diff
@@ -233,99 +239,148 @@ def predict(
             If the server encounters an error while handing the HTTP request.
 
         """
-        num_retries = 0
-        while num_retries < max_retries:
-            response = self._predict(x, compress)
-
-            if response.ok:
-                return _utils.body_to_json(response)
-            elif response.status_code in (
-                400,
-                502,
-            ):  # possibly error from the model back end
-                try:
-                    data = _utils.body_to_json(response)
-                except ValueError:  # not JSON response; 502 not from model back end
-                    pass
-                else:  # from model back end; contains message (maybe)
-                    # try to directly print message, otherwise line breaks appear as '\n'
-                    msg = data.get("message") or json.dumps(data)
-                    raise RuntimeError(
-                        "deployed model encountered an error: {}".format(msg)
-                    )
-            elif not (
-                response.status_code >= 500 or response.status_code in (404, 429)
-            ):  # clientside error
-                break
-
-            sleep = 0.3 * (2 ** (num_retries + 1))
-            print(
-                "received status {}; retrying in {:.1f}s".format(
-                    response.status_code, sleep
-                )
+        if always_retry_404 or always_retry_429:
+            warnings.warn(
+                "Deprecation warning: The `always_retry_404` and `always_retry_429`"
+                "parameters are deprecated and will removed in an upcoming version; "
+                "Status codes that should be retried are passed in the `retry_status`"
+                "parameter which includes 404 and 429 by default.",
+                category=FutureWarning,
+            )
+        prediction_with_id: Tuple[str, Any] = self.predict_with_id(
+            x=x,
+            compress=compress,
+            prediction_id=prediction_id,
+            max_retries=max_retries,
+            retry_status=retry_status,
+            backoff_factor=backoff_factor,
             )
-            time.sleep(sleep)
-            if (response.status_code == 404 and always_retry_404) or (  # model warm-up
-                response.status_code == 429 and always_retry_429
-            ):  # too many requests
-                num_retries = min(num_retries + 1, max_retries - 1)
-            else:
-                num_retries += 1
+        return prediction_with_id[1]
 
-        _utils.raise_for_http_error(response)
 
     def predict_with_id(
-        self,
-        x,
-        compress=False,
-        max_retries=5,
-        always_retry_404=True,
-        always_retry_429=True,
```

Comment:
API break? maybe we should keep just like we did for `predict`?

Reply:
These have been updated to `False` by default for backwards compatibility